### DISCUSSION TITLE
There may be some errors in the code of Kornia-MobileViT.

### DISCUSSION THREAD

**Discussion by yujieNC** [kornia/kornia#2151] [Potential Issue] [answered]
I think the following line is wrong. This line is not consistent with the description of MobileViT paper (https://arxiv.org/abs/2110.02178).
https://github.com/kornia/kornia/blob/84d76bfb0a2ea76a816978c891b6b77fff7ae367/kornia/contrib/vit_mobile.py#L194

The following line shows that the code of Kornia-MobileViT is based on https://github.com/chinhsuanwu/mobilevit-pytorch.
https://github.com/kornia/kornia/blob/84d76bfb0a2ea76a816978c891b6b77fff7ae367/kornia/contrib/vit_mobile.py#L209

But the corresponding line in https://github.com/chinhsuanwu/mobilevit-pytorch is:
https://github.com/chinhsuanwu/mobilevit-pytorch/blob/4df62dd3ab0c647e50eb082552c7c03b5e028cca/mobilevit.py#L156
 `x = rearrange(x, 'b d (h ph) (w pw) -> b (ph pw) (h w) d', ph=self.ph, pw=self.pw)`
where `rearrange` of einops is used.

@chinhsuanwu @edgarriba Are the following two lines equivalent? 

        x = x.view(b, self.ph * self.pw, (h // self.ph) * (w // self.pw), d)

        x = rearrange(x, 'b d (h ph) (w pw) -> b (ph pw) (h w) d', ph=self.ph, pw=self.pw)

**Comment by edgarriba** [comment]
That’s probably a mistake. Can you implement that line of code via permute ?

**Comment by chinhsuanwu** [comment, marked as answer]
Thanks for pointing out this issue. I have created a [PR](https://github.com/kornia/kornia/pull/2172) for this.
